src/Core/Config/PreloadPConfiguration.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Friendica\Core\Config;
+
+use Friendica\Model;
+
+/**
+ * This class implements the preload Time configuration, which will cache
+ * all user config values per call in a cache.
+ *
+ * Minimizes the number of database queries to retrieve configuration values at the cost of memory.
+ */
+class PreloadPConfiguration extends PConfiguration
+{
+	/** @var array */
+	private $config_loaded;
+
+	/**
+	 * @param Cache\PConfigCache   $configCache The configuration cache
+	 * @param Model\Config\PConfig $configModel The configuration model
+	 */
+	public function __construct(Cache\PConfigCache $configCache, Model\Config\PConfig $configModel)
+	{
+		parent::__construct($configCache, $configModel);
+		$this->config_loaded = [];
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * This loads all config values everytime load is called
+	 *
+	 */
+	public function load(int $uid, string $cat = 'config')
+	{
+		// Don't load the whole configuration twice
+		if (!empty($this->config_loaded[$uid])) {
+			return;
+		}
+
+		// If not connected, do nothing
+		if (!$this->configModel->isConnected()) {
+			return;
+		}
+
+		$config                    = $this->configModel->load($uid);
+		$this->config_loaded[$uid] = true;
+
+		// load the whole category out of the DB into the cache
+		$this->configCache->load($uid, $config);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function get(int $uid, string $cat, string $key, $default_value = null, bool $refresh = false)
```

Comment:
The `$refresh` parameter is neither necessary nor supported with the preload configuration. Any `set`/`remove` operation will update the cache anyway, making the new value immediately available to further `get` calls. I'm guessing this is an artifact from the legacy JIT configuration.

Reply:
Are you sure? I do find the `$refresh` parameter at the very first implementation of the preload config. 
https://github.com/friendica/friendica/blob/2a760bb770c015703567caa91054bcb3c9422fc0/src/Core/Config/PreloadConfigAdapter.php#L44-L56

https://github.com/friendica/friendica/blob/2a760bb770c015703567caa91054bcb3c9422fc0/src/Core/Config/PreloadPConfigAdapter.php#L48-L66

Reply:
And what about config values which are changed after the initial load, which should be "forced" to refresh? I think that's the same reason for this parameter as for JIT

Reply:
Ah, never mind, I forgot we have a daemon or a long-running worker. Both probably need the refresh feature even though I have no idea if they are using it (probably not).

Reply:
I found at least one occurence here:
https://github.com/friendica/friendica/blob/ff99a62584d256bef45a4053e2f6d505faf45c07/src/Util/ExAuth.php#L240